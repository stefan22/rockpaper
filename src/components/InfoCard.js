import React from 'react';
import PropTypes from 'prop-types';
import {withStyles} from '@material-ui/core/styles';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';
import timeout from '../images/timeout.png';

const styles = {
  card: {
    maxWidth: '100%',
    padding: '10px',
  },
  media: {
    height: '140px',
    maxWidth: '43%',
    margin: '0 auto',
  },
  span: {
    color: '#2196F3',
  }
};

function InfoCard(props) {
  const {classes} = props;
  return (
    <section className='col-4'>
      <Card className={classes.card}>
        <CardActionArea>
          <CardMedia
            className={classes.media}
            image={timeout}
            title={timeout}
          />
          <CardContent>
            <Typography gutterBottom variant="h5" component="h2">
              Game Rules:
            </Typography>
            <Typography component="p">
              • <span className={classes.span}>YOU</span> are
                Eric Cartman.<br />
              • Choose your <span className={classes.span}>OPPONENT
              </span> from your right <span className={classes.span}>FIRST
              </span>. <br />
              • <span className={classes.span}>PICK</span> your hand
              below to play the game.<br />
              • RESET game <span className={classes.span}>BEFORE
              </span> switching oponents. <br />

            </Typography>
          </CardContent>
        </CardActionArea>
        <CardActions>
          <Button
            onClick={props.handleGameReset}
            size="small" color="primary">
            Reset Game
          </Button>
        </CardActions>
      </Card>
    </section>
  );
}

InfoCard.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(InfoCard);
