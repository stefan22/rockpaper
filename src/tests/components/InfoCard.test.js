import React from 'react';
import {shallow} from 'enzyme';
import InfoCard from '../../components/InfoCard';

describe('InfoCard component', () => {
  it('should render InfoCard correctly', () => {
    const wrapper = shallow(<InfoCard />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should return shallow wrapper instance state null', () => {
    const wrapper = shallow(<InfoCard />);
    //const instance = wrapper.instance();
    //expect(instance['state']).toBe(null);
  });

});
