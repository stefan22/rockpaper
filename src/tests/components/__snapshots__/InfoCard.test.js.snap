// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoCard component should render InfoCard correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(InfoCard) />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "classes": Object {
        "card": "InfoCard-card-1",
        "media": "InfoCard-media-2",
        "span": "InfoCard-span-3",
      },
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "classes": Object {
          "card": "InfoCard-card-1",
          "media": "InfoCard-media-2",
          "span": "InfoCard-span-3",
        },
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
